### PR TITLE
Engine API: new/get_payload for Capella

### PR DESCRIPTION
**Pull request by terencechain**
This PR updates engine API implementations for newPayloadV2 and getPayloadV2 for Capella

### CODE REVIEW

File: beacon-chain/execution/engine_client.go [hunk 1]
```diff
@@ -93,13 +99,21 @@ func (s *Service) NewPayload(ctx context.Context, payload interfaces.ExecutionDa
...
-		return nil, errors.New("execution data must be an execution payload")
-	}
-	err := s.rpcClient.CallContext(ctx, result, NewPayloadMethod, payloadPb)
-	if err != nil {
-		return nil, handleRPCError(err)
+		payloadPb, ok := payload.Proto().(*pb.ExecutionPayload)
+		if !ok {
+			return nil, errors.New("execution data must be a Bellatrix or Capella execution payload")
+		}
+		err := s.rpcClient.CallContext(ctx, result, NewPayloadMethod, payloadPb)
```
Comment (nisdas) [inline beacon-chain/execution/engine_client.go:108]: can we do this via a switch statement by checking the payload type and then depending on that choose which payload method we want to call with ?

**Review by rauljordan — DISMISSED** [review]
LGTM after resolving the deepsource issues

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/execution/engine_client.go [hunk 2]
```diff
@@ -93,13 +99,28 @@ func (s *Service) NewPayload(ctx context.Context, payload interfaces.ExecutionDa
...
 	result := &pb.PayloadStatus{}
-	payloadPb, ok := payload.Proto().(*pb.ExecutionPayload)
-	if !ok {
-		return nil, errors.New("execution data must be an execution payload")
-	}
-	err := s.rpcClient.CallContext(ctx, result, NewPayloadMethod, payloadPb)
-	if err != nil {
-		return nil, handleRPCError(err)
+
+	switch payload.Proto().(type) {
```
Comment (alexmylonas) [inline beacon-chain/execution/engine_client.go:103]: Shouldn't this account for the EL?  To my understanding, the Capella execution payload is not accepted by Geth.

Getting constant warning that seems to cause the beacon sync to retreat.
```
Served engine_newPayloadV2               conn=127.0.0.1:39006 reqid=9 duration="17.91µs"   err="the method engine_newPayloadV2 does not exist/is not available"
```

And on Prysm side:
```
level=warning msg="Skip processing batched blocks" error="could not process block in batch: method not found: received an undefined execution engine error" prefix=initial-sync
```
I 'll be happy to open an issue if you need more info!

Network: Goerli-Prater
Geth: v1.10.26
Prysm: v3.2

Comment (nisdas) [reply beacon-chain/execution/engine_client.go:103]: Which network is this ? if this is goerli, you will need to update to the latest version for each client

Comment (alexmylonas) [reply beacon-chain/execution/engine_client.go:103]: I am on Goerli and just saw the [prysm release notes](https://github.com/prysmaticlabs/prysm/releases/tag/v3.2.2) about it. 

And on the [geth](https://github.com/ethereum/go-ethereum/releases/tag/v1.11.3) side as well.

Thanks for the quick response! 🙏  Resolving comment.

**Pull request merged**